scripts/build-validate.js

```diff
@@ -20,6 +20,16 @@ function validateStandardBuild() {
   assert.ok(fs.existsSync('./dist/index.es.js'));
   assert.ok(fs.existsSync('./dist/styles.css'));
 
+  // Assert it uses named exports rather than properties from the React default
+  // export to help tree-shaking.
+  // React.createElement and React.Fragment are the allowed exceptions
```

Comment:
Curious about why we'd allow exceptions, can you say more?

Reply:
This tests against the compiled file. This means that the compiled file may include them and that folks may end up writing them manually (though 99% of the time won't need to as you should just use JSX). We need these exceptions because these are what the JSX transform uses. When we write:

```js
const x = <div>Hi <>you</></div>
```

it gets compiled to:

```
const x = /*#__PURE__*/React.createElement("div", null, "Hi ", /*#__PURE__*/React.createElement(React.Fragment, null, "you"));
```

--- 

These asserts accept a final arg where you can write a description. I'll flesh that out